polyfill/lib/ecmascript.mjs

```diff
@@ -95,12 +95,16 @@ const ToPositiveInteger = (value, property) => {
   }
   return value;
 };
-const ToIntegerNoFraction = (value) => {
+const ToIntegerNoTruncate = (value) => {
   value = ES.ToNumber(value);
+  if (NumberIsNaN(value)) return 0;
+  if (!NumberIsFinite(value)) {
+    throw new RangeError('infinity is out of range');
+  }
   if (!ES.IsInteger(value)) {
     throw new RangeError(`unsupported fractional value ${value}`);
   }
-  return value;
+  return ES.ToInteger(value);
```

Comment:
If this call does something - and it might for `-0` - this should have a comment